docs/proposal/20250324-rosa-iam-roles.md

```diff
@@ -0,0 +1,226 @@
+# Propose ROSA HCP IAM Account Roles, IAM Operator Roles, and OIDC Configuration as CRD in CAPA
+
+## Table of Contents
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Problem Statement](#problem-statement)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories](#user-stories)
+  - [Implementation Details](#implementation-details)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Alternatives](#alternatives)
+- [Implementation Plan](#implementation-plan)
+- [Resources](#resources)
+
+## Summary
+The ROSA HCP in CAPA (Cluster API Provider AWS) project currently lacks built-in support for managing IAM account roles, IAM operator roles, and OIDC configuration as Custom Resource Definitions (CRDs). This proposal suggests integrating the [ROSA library](https://github.com/openshift/rosa) to extend ROSA-HCP's functionality, enabling declarative IAM and OIDC management.
+
+## Motivation
+### Problem Statement
+Managing ROSA HCP IAM roles and OIDC configurations manually can be complex and error-prone. ROSA-HCP in CAPA lacks an automated mechanism to provision and reconcile these IAM resources, making it challenging for OpenShift ROSA HCP clusters deployed via CAPA to comply with security best practices and AWS integration requirements.
+
+### Goals
+- Introduce ROSARoleConfig CRD for ROSA HCP IAM account roles, IAM operator roles, and OIDC configuration in CAPA.
+- Enable automated reconciliation of these resources using the ROSA library.
+- Simplify cluster lifecycle management by aligning IAM/OIDC management with Kubernetes declarative paradigms.
+- Reduce operational overhead for users deploying OpenShift ROSA HCP clusters on AWS via CAPA.
+
+### Non-Goals
+- Replacing existing IAM management workflows outside of CAPA.
```

Comment:
Did you mean to write `outside of ROSA` here?

Reply:
ya , I will fix that.